js/models/adaptModel.js

```diff
@@ -468,6 +468,8 @@ export default class AdaptModel extends LockingModel {
 
     children.models.forEach(child => {
```

Comment:
When is children.models getting populated with undefined? `[undefined]`

Reply:
Yeah it is concerning. I have not been able to pin it down while debugging. The sheer number of loops through `setChildren` (100's?) has made it difficult, and any conditional debugging (checking for undefined value for example) doesn't seem to kick in as I expect, so something strange going on.

I figured this would be a beneficial early return check that gets us past the problem of the course falling over

Reply:
@oliverfoster Any further thoughts on this?

Reply:
Sorry mate, I missed this comment.